src/components/ReminderCtaButton.tsx

```diff
@@ -1,123 +1,106 @@
 import React, { useState } from 'react';
-import { css, ThemeProvider } from '@emotion/react';
-import { neutral, brandAlt, news, body, space } from '@guardian/source-foundations';
+import { css, SerializedStyles, ThemeProvider } from '@emotion/react';
+import { body, space } from '@guardian/source-foundations';
 import { Button } from '@guardian/source-react-components';
 
-import { buildReminderFields, createReminder, ReminderStage } from '../logic/reminders';
-import type { InteractiveButtonStatus } from '../logic/types';
+import {
+    buildReminderFields,
+    createReminder,
+    ReminderStage,
+    ReminderComponent,
+} from '../logic/reminders';
+import type { InteractiveButtonStatus, Extras } from '../logic/types';
 
 import { FetchEmail } from '../types/dcrTypes';
 import { TrackClick } from '../utils/tracking';
-import { LoadingDots } from './CtaLoadingDotsAnimation';
+import { buildLoadingDots } from './CtaLoadingDotsAnimation';
+import { contributionsTheme, getColors, ColorStylesData } from '../styles/colorData';
 
-// Custom theme for Button/LinkButton
-// See also `tertiaryButtonOverrides` below.
-const buttonStyles = {
-    textPrimary: neutral[7],
-    backgroundPrimary: brandAlt[400],
-    backgroundPrimaryHover: brandAlt[300],
-    textSecondary: neutral[7],
-    backgroundSecondary: neutral[93],
-    backgroundSecondaryHover: neutral[86],
-    borderSecondary: neutral[86],
+const defaultButtonColors: ColorStylesData = {
+    styleReminderButton: '#121212',
+    styleReminderButtonBackground: '#f6f6f6;',
+    styleReminderButtonHover: '#dcdcdc',
 };
 
-const contributionsTheme = {
-    button: buttonStyles,
-    link: buttonStyles,
-};
-
-const styles = {
-    buttonWrapperStyles: css`
-        margin: ${space[4]}px ${space[2]}px ${space[1]}px 0;
-        display: flex;
-        flex-direction: column;
-        flex-wrap: wrap;
-        justify-content: flex-start;
-        align-items: flex-start;
-        max-width: 50%;
-    `,
-    buttonMargins: css`
-        margin: ${space[1]}px ${space[2]}px ${space[1]}px 0;
-    `,
-    button: css`
-        background-color: ${news[400]};
-        color: ${neutral[100]};
-        &:hover {
-            background-color: ${news[400]};
-        }
-    `,
-    thankYouText: css`
-        ${body.medium({ fontWeight: 'bold' })};
-        color: ${neutral[0]};
-        margin-top: ${space[3]}px;
-    `,
-    remindMeButtonOverrides: css`
-        background-color: transparent !important;
-        color: ${neutral[7]} !important;
-
-        :hover {
-            background-color: ${neutral[86]} !important;
-        }
-    `,
-    smallPrint: css`
-        color: ${neutral[0]};
-        margin-top: ${space[2]}px;
-        ${body.small()};
-    `,
+const getButtonStyles = (userVals: Extras, defaults: ColorStylesData) => {
```

Comment:
I wonder if we could be more descriptive/specific with the types of these args. `Extras` is very generic and doesn't provide much information about what might be available. My understanding is `getColors` takes some colours which _might_ be provided and some defaults which _will_ be provided and comes up with a new instance of `ColorStylesData` which is guaranteed to have specific fields set (if we can't get the colour from the user provided values then fall back on the defaults).

What if the type of the user provided styles was called something like `UserDefinedColorStylesData` and it looked something like this (clipped for brevity):

```
export interface UserDefinedColorStylesData {
    styleBackground?: string;
    styleHeader?: string;
    styleBody?: string;
}
```

So any of those fields may be there, but could also be missing. I think this tells us more about the possible user provided styles than `Extras` does. Then we could define `ColorStylesData` as the type which is guaranteed to have all these fields:

```
type ColorStylesData = Required<UserDefinedColourStylesData>;
```

Then the type of `getColors` becomes: `(userVals: UserDefinedColorStylesData, defaults: ColorStylesData): ColorStylesData`.

Reply:
Will take this suggestion forward. We can pair if I run into TS issues?

If we can get rid of `Extras` ... I wouldn't cry. I think there's another type with the same definition somewhere.

Reply:
> We can pair if I run into TS issues?

Definitely! 👍

Reply:
I've given it a go. Discuss tomorrow?